packages/flutter/lib/src/material/popup_menu.dart

```diff
@@ -26,20 +27,36 @@ const double _kMenuVerticalPadding = 8.0;
 const double _kMenuWidthStep = 56.0;
 const double _kMenuScreenPadding = 8.0;
 
-class PopupMenuItem<T> extends StatelessComponent {
+abstract class PopupMenuEntry<T> extends StatelessComponent {
+  PopupMenuEntry({ Key key }) : super(key: key);
+
+  double get height;
+  T get value => null;
+  bool get enabled => true;
+}
+
+class PopupMenuDivider<T> extends PopupMenuEntry<T> {
```

Comment:
I think `PopupMenuDivider` should probably explicitly extend `PopupMenuEntry<dynamic>`. Then, if I'm not mistaken, you'll be able to use it in a `PopupMenuEntry<T>` list (for any value of T).